plugins/notifications-node/src/extensions.ts

```diff
@@ -15,25 +15,58 @@
  */
 import { createExtensionPoint } from '@backstage/backend-plugin-api';
 import { Notification } from '@backstage/plugin-notifications-common';
+import { NotificationSendOptions } from './service';
 
 /**
  * @public
  */
 export interface NotificationProcessor {
   /**
-   * Decorate notification before sending it
+   * Human-readable name of this processor like Email, Slack, etc.
+   */
+  getName(): string;
+
+  /**
+   * Process the notification options.
+   *
+   * This can be used to modify the options before sending the notification or even sending the notification to
+   * external services. This function is called only once for each notification before processing it.
+   *
+   * @param options - The original options to send the notification
+   */
+  processOptions?(
```

Comment:
Honestly, the name `Options` does not tell me much.
Is it kind of decorator?

Can we document an example when this is useful and why preProcess is not enough?

Reply:
This can be used if you want to resolve the receivers yourself for example or want to send the notification only once (preProcess/postProcess are called for each user separately while this is called only once for each notification POST request).

Reply:
Can you please update the comments/documentation? It will truly simplify usage.

Reply:
Hm, I don't know how to make it more clear. It already says:

```
This function is called only once for each notification before processing it.
```

Reply:
Added some docs, hope it makes more sense now 👍